server.js

```diff
@@ -2,37 +2,113 @@ import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import booksData from './data/books.json'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project_mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
+// Defines the port the app will run on. Can be 
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
+const port = process.env.PORT || 9001
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+// Mongoose model of books
+const Book = mongoose.model('Book', {
+  bookID: Number,
+  title: String,
+  authors: String,
+  average_rating: Number,
+  isbn: Number,
+  isbn13: Number,
+  language_code: String,
+  num_pages: Number,
+  ratings_count: Number,
+  text_reviews_count: Number
+})
+
+if(process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Book.deleteMany({})
+
+    booksData.forEach((bookData) => {
+      new Book(bookData).save()
+    })
+  }
+  seedDatabase()
+}
+
+// Start defining the routes
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Books API')
+})
+
+// Return the array with all the book objects
+app.get('/books', (req, res) => {
+  const { language } = req.query
+  const queryString = req.query.title 
+  const queryRegex = new RegExp(queryString, "i")
+
+
+// Using query for language code
+// For example, http://localhost:9001/books?language=eng will return books with English language code
+  if(language) {
+   Book.find({'language_code': language})
+     .then((results) => {
+       console.log('Found')
+       res.json(results)
+      }) .catch((err) => {
+        console.log('Error ' + err)
+        res.json({message: 'Cannot find book', err: err}) 
+    })
+  }
+
+
+// Using query for title
+// For example, http://localhost:9001/books?title=Harry will return books with 'Harry' in the title
+// Also, the books are sorted by average rating from the highest to lowest
+  Book.find({'title': queryRegex})
+    .sort({'average_rating': -1})
+    .then((results) => {
+      console.log('Found')
+      res.json(results)
+    }) .catch((err) => {
+       console.log('Error ' + err)
+       res.json({message: 'Cannot find book', err: err}) 
+    })
+})
```

Comment:
Usually it's a good idea to avoid too many/multiple exit points in a function.  Is there a way to restructure the function to only send responses in two cases (success/failure)?